File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        Map<String, List<String>> map = new HashMap<>();

        for(String str : strs){

            char[] charArr = str.toCharArray();
            Arrays.sort(charArr);
            String key = new String(charArr);

            if(map.containsKey(key)){
                map.get(key).add(str);
            }else{
                map.put(key, new ArrayList<>());
                map.get(key).add(str);
            }
        }

        return new ArrayList<>(map.values());
        
    }
}

File: 0049-group-anagrams/0049-group-anagrams.rs
use std::collections::HashMap;


impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        
        let mut map: HashMap<String, Vec<String>> = HashMap::new();
        let mut result: Vec<Vec<String>> = Vec::new();

        for word in strs.iter() {

            let mut bytes = word.as_bytes().to_vec();
            bytes.sort();
            let sorted_word = String::from_utf8(bytes).unwrap();

            if map.contains_key(&sorted_word) {
                map.get_mut(&sorted_word).unwrap().push(word.clone());
            }else{
                map.insert(String::from(sorted_word), vec![String::from(word.clone())]);
            }
        }

        for (key, value) in map.iter() {
            result.push(value.clone());
        }

        return result;

    }
}